youtube_dl/extractor/tiktok.py

```diff
@@ -1,138 +1,252 @@
-            try_get(data, lambda x: x['statistics']['comment_count']))
-        repost_count = int_or_none(try_get(
-            data, lambda x: x['statistics']['share_count']))
-
-        aweme_id = data['aweme_id']
+        formats.append({
+            'url': video_info.get('video_url') or self._og_search_video_url(webpage),
+            'ext': 'mp4',
+            'height': video_info.get('height'),
+            'width': video_info.get('width'),
+        })
 
         return {
-            'id': aweme_id,
-            'title': uploader or aweme_id,
+            'artist': track_info.get('artist'),
+            'channel_id': author_info.get('channel_id'),
+            'channel_url': 'https://www.tiktok.com/@{}'.format(author_info.get('uploader')),
+            'comment_count': video_info.get('comment_count'),
+            'creator': author_info.get('creator'),
             'description': description,
-            'thumbnail': thumbnail,
-            'uploader': uploader,
-            'timestamp': timestamp,
-            'comment_count': comment_count,
-            'repost_count': repost_count,
+            'duration': video_info.get('duration'),
             'formats': formats,
+            'height': video_info.get('height'),
+            'id': video_info.get('id'),
+            'like_count': video_info.get('like_count'),
+            'playlist_title': share_info.get('title'),
+            'playlist_uploader': author_info.get('uploader'),
+            'playlist_uploader_id': author_info.get('uploader_id'),
+            'repost_count': video_info.get('repost_count'),
+            'release_date': date,
+            'tags': tags,
+            'thumbnail': video_info.get('thumbnail'),
+            'thumbnails': thumbnails,
+            'timestamp': int(video_info.get('timestamp')),
+            'title': share_info.get('title') or self._og_search_title(webpage),
+            'track': track_info.get('track'),
+            'track_id': track_info.get('track_id'),
+            'upload_date': date,
+            'uploader': author_info.get('uploader'),
+            'uploader_id': author_info.get('uploader_id'),
+            'uploader_url': 'https://www.tiktok.com/@{}'.format(author_info.get('uploader')),
+            'webpage_url': self._og_search_url(webpage),
+            'width': video_info.get('width')
         }
 
 
 class TikTokIE(TikTokBaseIE):
     _VALID_URL = r'''(?x)
-                        https?://
-                            (?:
-                                (?:m\.)?tiktok\.com/v|
-                                (?:www\.)?tiktok\.com/share/video
-                            )
-                            /(?P<id>\d+)
-                    '''
+                 https?://
+                 (?:
+                    (?:www|m)\.
+                    (?:tiktok.com)\/
+                    (@(?P<username>[\w\.]+))?
+                    (?:v|video|embed|trending)?(?:\/)?
+                    (?:video)?(?:\/)?
+                    (?:\?shareId=)?
```

Comment:
This is total mess. Should not match https://www.tiktok.com/@leenabhushanvideo6748451240264420610, https://www.tiktok.com/embed/video/6567659045795758085 and so on.